persist-cli/src/main/java/io/ballerina/persist/pluralize/Pluralize.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.persist.pluralize;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Transforms english words from singular to plural form.
+ */
+public class Pluralize {
+
+    public static String getResourceNameInPlural(String resourceName) {
+        for (String rule : Rules.UNCOUNTABLE_RULES) {
+            if (Pattern.matches(rule, resourceName)) {
+                return resourceName;
+            }
+        }
+        for (String[] irregularRule: Rules.IRREGULAR_RULES) {
+            if (irregularRule[0].equals(resourceName)) {
+                return irregularRule[1];
+            }
+        }
+        for (String[] pluralizationRule: Rules.PLURALIZATION_RULES) {
+            Matcher matcher = Pattern.compile(pluralizationRule[0]).matcher(resourceName);
+            if (matcher.find()) {
+                return matcher.replaceFirst(pluralizationRule[1]);
+            }
+        }
+        return resourceName;
```

Comment:
Have you checked average runtime for this code segment? There are lot of pattern matches, how long does it take in average. IMO better to do a load test with atleast 100 odd words, to be on the safe side.

This can be done as a future improvement too.

Reply:
Added a unit test with 100 words. It took 62-74ms to perform 100 words in the plural. 
And in the description, added a table with some individual words and time to finish for making plural.

Reply:
Ack